src/sparseml/transformers/export.py

```diff
@@ -233,6 +245,56 @@ def export_transformer_to_onnx(
     return onnx_file_path
 
 
+def create_deployment_folder(
+    training_directory: str,
+    onnx_file_name: str = MODEL_ONNX_NAME,
+    deployment_files: Set[str] = DEPLOYMENT_FILES,
+):
+    """
+    Sets up the deployment directory i.e. copies over the complete set of files
+    that are required to run the transformer model in the inference engine
+
+    :param training_directory: path to directory where model files, tokenizers,
+        and configs are saved. Exported ONNX model is also expected to be there
+    :param onnx_file_name: Name for exported ONNX file in the model directory.
+    :param deployment_files: The set of files that are expected to be present in
+        to the deployment folder once this function terminates.
+    :return: path to the valid deployment directory
+    """
+
+    if onnx_file_name != MODEL_ONNX_NAME:
+        # replace the default onnx name with the custom one
+        deployment_files.remove(MODEL_ONNX_NAME)
+        deployment_files.update(onnx_file_name)
+
+    if training_directory.split("/")[-1] != "training":
+        _LOGGER.warning(
+            "Expected to receive path to the training directory, "
+            f"but received path to {training_directory.split('/')[1]} directory/file"
+        )
+
+    model_root_dir = os.path.dirname(training_directory)
+    deployment_folder_dir = os.path.join(model_root_dir, "deployment")
+    if os.path.isdir(deployment_folder_dir):
+        shutil.rmtree(deployment_folder_dir)
+    os.makedirs(deployment_folder_dir)
+
+    for file_name in deployment_files:
+        expected_file_path = os.path.join(training_directory, file_name)
+        deployment_file_path = os.path.join(deployment_folder_dir, file_name)
+        if not os.path.exists(expected_file_path):
+            raise ValueError(
+                f"Attempting to copy {file_name} file from {expected_file_path},"
+                f"but the file does not exits. Make sure that {training_directory} "
+                f"contains following files: {deployment_files}"
+            )
+        shutil.copyfile(expected_file_path, deployment_file_path)
```

Comment:
In terms of files that aren't a part of the training output (i.e. the .onnx file), what are your thoughts on moving instead of copying? Besides the additional memory overhead, I think it might be confusing to the user when the .onnx file gets generated in two different locations